torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -289,7 +290,13 @@ def set_backend_pattern_config(self, config: BackendPatternConfig) -> BackendCon
         Set the config for an pattern that can be run on the target backend.
         This overrides any existing config for the given pattern.
         """
-        self.configs[config.pattern] = config
+        if config.pattern is not None:
+            pattern = config.pattern
+        elif config._pattern_complex_format is not None:
+            pattern = config._pattern_complex_format
+        else:
+            raise ValueError("Expected either 'pattern' or 'pattern_complex_format' to be set")
+        self._config_dict[pattern] = config
```

Comment:
I'm wondering if it would be better to convert the pattern to complex format when we set it, so that internally we don't need to handle multiple formats

Reply:
also in this case, using `pattern` as a key might be confusing I think? since we are not sure what format this pattern is using given the pattern itself

Reply:
I also considered that, but I think it would be even more confusing for the user if they try to read the configs afterwards (to find that their patterns suddenly changed). Also eventually I think we want a simpler matcher, at that point we'll want to use the simple format as the internal format

Reply:
> also in this case, using `pattern` as a key might be confusing I think? since we are not sure what format this pattern is using given the pattern itself

Ok, for this I can separate this into two dicts, one for each format. The key here is mainly just for detecting duplicates, but I agree it may be confusing to consumers of this dict.

Reply:
Ok, I convert the pattern when we set it now so we only store the complex format. I also renamed the variable from `_config_dict` to `_pattern_complex_format_to_config` to better reflect that.